scipy/spatial/transform/_rotation.pyx

```diff
@@ -2806,54 +2833,237 @@ cdef class Rotation:
+                Control, and Dynamics, Vol. 20, No. 5, 1997, pp. 916-922.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Best align two sets of vectors with the Kabsch algorithm, where there
+        is noise on the last two vector measurements of the ``b`` set:
+
+        >>> a = [[0, 1, 0], [0, 1, 1], [0, 1, 1]]
+        >>> b = [[1, 0, 0], [1, 1.1, 0], [1, 0.9, 0]]
+        >>> rot, rssd, sens = R.align_vectors(a, b, return_sensitivity=True)
+        >>> rot.as_matrix()
+        array([[0., 0., 1.],
+               [1., 0., 0.],
+               [0., 1., 0.]])
+
+        When we apply the rotation to ``b``, we get vectors close to ``a``:
+
+        >>> rot.apply(b)
+        array([[0. , 1. , 0. ],
+               [0. , 1. , 1.1],
+               [0. , 1. , 0.9]])
+
+        The error for the first vector is 0, and for the last two the error is
+        magnitude 0.1. The `rssd` is the square root of the sum of the
+        weighted squared errors, and the default weights are all 1, so in this
+        case the `rssd` is calculated as
+        ``sqrt(1 * 0**2 + 1 * 0.1**2 + 1 * (-0.1)**2) = 0.141421356237308``
+
+        >>> a - rot.apply(b)
+        array([[ 0., 0.,  0. ],
+               [ 0., 0., -0.1],
+               [ 0., 0.,  0.1]])
+        >>> np.sqrt(np.sum(np.ones(3) @ (a - rot.apply(b))**2))
+        0.141421356237308
+        >>> rssd
+        0.141421356237308
+
+        The sensitivity matrix for this example is as follows:
+
+        >>> sens
+        array([[0.2, 0. , 0.],
+               [0. , 1.5, 1.],
+               [0. , 1. , 1.]])
+
+        Special case 1: Find a minimum rotation between single vectors:
+
+        >>> a = [1, 0, 0]
+        >>> b = [0, 1, 0]
+        >>> rot, _ = R.align_vectors(a, b)
+        >>> rot.as_matrix()
+        array([[0., 1., 0.],
+               [-1., 0., 0.],
+               [0., 0., 1.]])
+        >>> rot.apply(b)
+        array([1., 0., 0.])
+
+        Special case 2: One infinite weight. Here we find a rotation between
+        primary and secondary vectors that can align exactly:
+
+        >>> a = [[0, 1, 0], [0, 1, 1]]
+        >>> b = [[1, 0, 0], [1, 1, 0]]
+        >>> rot, _ = R.align_vectors(a, b, weights=[np.inf, 1])
+        >>> rot.as_matrix()
+        array([[0., 0., 1.],
+               [1., 0., 0.],
+               [0., 1., 0.]])
+        >>> rot.apply(b)
+        array([[0., 1., 0.],
+               [0., 1., 1.]])
 
+        Here the secondary vectors must be best-fit:
+
+        >>> b = [[1, 0, 0], [1, 2, 0]]
+        >>> rot, _ = R.align_vectors(a, b, weights=[np.inf, 1])
+        >>> rot.as_matrix()
+        array([[0., 0., 1.],
+               [1., 0., 0.],
+               [0., 1., 0.]])
+        >>> rot.apply(b)
+        array([[0., 1., 0.],
+               [0., 1., 2.]])
+        """
+        # Check input vectors
+        a = np.array(a, dtype=float)
```

Comment:
I think here we should call `atleast_2d`  right here.

Reply:
This needed a little rework to make the input checks report out the actual user input, but the logic makes sense now.